### PR TITLE
[DOCU-2121] Add Alternate OpenAPI Renderer page

### PR DESCRIPTION
**Pull request by falondarville**
### Summary

Add instructions for how to render Stoplight and Redoc from OpenAPI spec docs in Dev Portal. Also add `spec-renderer.html` file as it's longer and the user will need to copy and paste its entirety. It'll be easier for them to do that from a separate file.

### Reason

Addresses [DOCU-2121](https://konghq.atlassian.net/browse/DOCU-2121)

### Testing

https://deploy-preview-3700--kongdocs.netlify.app/gateway/2.7.x/developer-portal/theme-customization/alternate-openapi-renderer/

FYI I believe that the `spec-renderer.html` file is not linking because it uses the variable `latest`. We may have to manually approve the links on this one.

### CODE REVIEW

**Review by lena-larionova — COMMENTED** [review]
Re: the linking to the spec - it looks like we don't generate /latest/ URLs for files that aren't `.md`. That's not great. I just checked with the `statsd.rules.yaml` file: 

* This URL is broken: https://docs.konghq.com/gateway/latest/statsd.rules.yaml 
* This URL works: https://docs.konghq.com/gateway/2.7.x/statsd.rules.yaml

Tl;dr, this is a bigger issue that you don't need to fix here, but obviously we still need a working link. 

My suggestion is to create a separate folder in the `/app` directory instead and call it `examples` (so your path will be `/app/examples/<filename>.html`). Or maybe `code-snippets` instead of `examples`? What do you think?

**Comment by falondarville** [comment]
I updated the following:
1. Create `code-snippets` folder in `/app`
2. Move `spec-renderer.html` and `statsd.rules.yaml` to `code-snippets` folder
3. Update the links to the two aforementioned files 
4. Add the same content to 2.8.x

File: app/gateway/2.7.x/developer-portal/theme-customization/alternate-openapi-renderer.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+Through Kong Manager, use the Dev Portal [Editor](/gateway/latest/developer-portal/using-the-editor) to customize how your API specs render in your Dev Portal.
+
+To do this, add some custom code and update your config, as follows:
+
+1. In Kong Manager, navigate to the Dev Portal Editor from the left sidebar.
+
+1. Navigate to **Themes** > **base** > **layout** > **system** > **`spec.renderer.html`**.
+
+1. In `spec.renderer.html`, replace all of the current content with the content from this [spec-renderer.html](../../../code-snippets/spec-renderer.html) file. Doing so adds options for [Stoplight](https://meta.stoplight.io/docs/platform/ZG9jOjIwNjk2MQ-welcome-to-the-stoplight-docs) and [Redoc](https://github.com/Redocly/redoc) layouts.
```
Comment (lena-larionova) [inline app/gateway/2.7.x/developer-portal/theme-customization/alternate-openapi-renderer.md:14]:
```suggestion
1. In `spec.renderer.html`, replace all of the current content with the content from this [spec-renderer.html](/code-snippets/spec-renderer.html) file. Doing so adds options for [Stoplight](https://meta.stoplight.io/docs/platform/ZG9jOjIwNjk2MQ-welcome-to-the-stoplight-docs) and [Redoc](https://github.com/Redocly/redoc) layouts.
```
The URL is relative to the `app` folder, so you just need this.

File: app/gateway/2.7.x/vitals/vitals-prometheus-strategy.md [hunk 2]
```diff
@@ -78,7 +78,7 @@ socket support which are not supported in the public project.
 StatsD exporter needed to configured with a set of mapping rules to translate
 the StatsD UDP events to Prometheus metrics. A default set of mapping rules can
 be downloaded at
-[statsd.rules.yaml](/gateway/{{page.kong_version}}/statsd.rules.yaml).
+[statsd.rules.yaml](../../../code-snippets/statsd.rules.yaml/statsd.rules.yaml).
```
Comment (lena-larionova) [inline app/gateway/2.7.x/vitals/vitals-prometheus-strategy.md:81]:
```suggestion
[statsd.rules.yaml](/code-snippets/statsd.rules.yaml).
```

File: app/gateway/2.8.x/developer-portal/theme-customization/alternative-openapi-renderer.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+Through Kong Manager, use the Dev Portal [Editor](/gateway/latest/developer-portal/using-the-editor) to customize how your API specs render in your Dev Portal.
+
+To do this, add some custom code and update your config, as follows:
+
+1. In Kong Manager, navigate to the Dev Portal Editor from the left sidebar.
+
+1. Navigate to **Themes** > **base** > **layout** > **system** > **`spec.renderer.html`**.
+
+1. In `spec.renderer.html`, replace all of the current content with the content from this [spec-renderer.html](../../../code-snippets/spec-renderer.html) file. Doing so adds options for [Stoplight](https://meta.stoplight.io/docs/platform/ZG9jOjIwNjk2MQ-welcome-to-the-stoplight-docs) and [Redoc](https://github.com/Redocly/redoc) layouts.
```
Comment (lena-larionova) [inline app/gateway/2.8.x/developer-portal/theme-customization/alternative-openapi-renderer.md:14]:
```suggestion
1. In `spec.renderer.html`, replace all of the current content with the content from this [spec-renderer.html](/code-snippets/spec-renderer.html) file. Doing so adds options for [Stoplight](https://meta.stoplight.io/docs/platform/ZG9jOjIwNjk2MQ-welcome-to-the-stoplight-docs) and [Redoc](https://github.com/Redocly/redoc) layouts.
```

File: app/gateway/2.7.x/developer-portal/theme-customization/alternate-openapi-renderer.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+---
+title: Alternate OpenAPI Renderer
+badge: enterprise
+---
+
+Through Kong Manager, use the Dev Portal [Editor](/gateway/latest/developer-portal/using-the-editor) to customize how your API specs render in your Dev Portal.
```
Comment (lena-larionova) [inline app/gateway/2.7.x/developer-portal/theme-customization/alternate-openapi-renderer.md:6]:
```suggestion
Through Kong Manager, use the Dev Portal [Editor](/gateway/{{page.kong_version}}/developer-portal/using-the-editor) to customize how your API specs render in your Dev Portal.
```

File: app/gateway/2.8.x/developer-portal/theme-customization/alternative-openapi-renderer.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+---
+title: Alternate OpenAPI Renderer
+badge: enterprise
+---
+
+Through Kong Manager, use the Dev Portal [Editor](/gateway/latest/developer-portal/using-the-editor) to customize how your API specs render in your Dev Portal.
```
Comment (lena-larionova) [inline app/gateway/2.8.x/developer-portal/theme-customization/alternative-openapi-renderer.md:6]:
```suggestion
Through Kong Manager, use the Dev Portal [Editor](/gateway/{{page.kong_version}}/developer-portal/using-the-editor) to customize how your API specs render in your Dev Portal.
```

File: app/gateway/2.7.x/vitals/vitals-prometheus-strategy.md (hunk 2 above)
Comment (lena-larionova) [reply app/gateway/2.7.x/vitals/vitals-prometheus-strategy.md:81]: Can you make this change to the 2.8 version also?

**Review by lena-larionova — CHANGES_REQUESTED** [review]

**Review by lena-larionova — COMMENTED** [review]
Looks like the filename isn't quite right in 2.8.x: `alternative-openapi-renderer.md`

All the links are looking for `alternate-openapi-renderer.md`

**Review by lena-larionova — APPROVED** [review]
LGTM!
